packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -224,23 +269,59 @@ class SnackBar extends StatelessWidget {
     ];
     if (action != null) {
       children.add(ButtonTheme.bar(
-        padding: const EdgeInsets.symmetric(horizontal: _kSnackBarPadding),
+        padding: EdgeInsets.symmetric(horizontal: snackBarPadding),
         textTheme: ButtonTextTheme.accent,
         child: action,
       ));
     } else {
-      children.add(const SizedBox(width: _kSnackBarPadding));
+      children.add(SizedBox(width: snackBarPadding));
     }
     final CurvedAnimation heightAnimation = CurvedAnimation(parent: animation, curve: _snackBarHeightCurve);
-    final CurvedAnimation fadeAnimation = CurvedAnimation(parent: animation, curve: _snackBarFadeCurve, reverseCurve: const Threshold(0.0));
-    Widget snackbar = SafeArea(
+    final CurvedAnimation fadeInAnimation = CurvedAnimation(parent: animation, curve: _snackBarFadeInCurve);
+    final CurvedAnimation fadeOutAnimation = CurvedAnimation(
+      parent: animation,
+      curve: _snackBarFadeOutCurve,
+      reverseCurve: const Threshold(0.0),
+    );
+
+    Widget snackBar = SafeArea(
       top: false,
+      bottom: !isFloatingSnackBar,
       child: Row(
         children: children,
         crossAxisAlignment: CrossAxisAlignment.center,
       ),
     );
-    snackbar = Semantics(
+
+    final double elevation = this.elevation ?? snackBarTheme.elevation ?? 6.0;
+    final Color backgroundColor = this.backgroundColor ?? snackBarTheme.backgroundColor ?? _kSnackBarBackgroundColor;
+    final ShapeBorder shape = this.shape
+        ?? snackBarTheme.shape
+        ?? (isFloatingSnackBar ? RoundedRectangleBorder(borderRadius: BorderRadius.circular(4.0)) : null);
+
+    snackBar = Material(
+      shape: shape,
+      elevation: elevation,
+      color: backgroundColor,
+      child: Theme(
+        data: darkTheme,
+        child: mediaQueryData.accessibleNavigation
+            ? snackBar
+            : FadeTransition(
+                opacity: fadeOutAnimation,
+                child: snackBar,
+              ),
+      ),
+    );
+
+    if (isFloatingSnackBar) {
```

Comment:
Wouldn't `Padding(padding: const EdgeInsets.fromLTRB(15.0, 5.0, 15.0, 10.0), child: snackBar)` have the same effect (it's a little simpler)?